src/bulk/common.ts

```diff
@@ -544,7 +547,8 @@ function mergeBatchResults(
   }
 
   if (result.writeConcernError) {
-    bulkResult.writeConcernErrors.push(new WriteConcernError(result.writeConcernError));
+    const { code, errmsg, errInfo } = result.writeConcernError;
+    bulkResult.writeConcernErrors.push(new WriteConcernError({ code, errmsg, errInfo }));
```

Comment:
I think we have to change this back to `result.writeConcernError`, we might be losing properties in this destructuring.
We can make the constructor accept a more flexible type, it's still good that we have a better typed `WriteConcernError` class, but incase there are more properties in certain server versions we should have them automatically pass through here.